text/0059-geo-search.md

```diff
@@ -0,0 +1,232 @@
+- Title: Geo-search
+- Start Date: 2021-08-02
+- Specification PR: [#59](https://github.com/meilisearch/specifications/pull/59)
+- Discovery Issue: [#42](https://github.com/meilisearch/product/issues/42)
+- MeiliSearch Tracking-issues:
+
+# Geo-search
+
+## 1. Functional Specification
+
+### I. Summary
+
+The purpose of this specification is to add a first iteration of the **geo-search** feature to give geo-filtering and geo-sorting capabilities at search time.
+
+#### Summary Key points
+
+- Documents MUST have a `_geo` reserved object to be geo-searchable.
+- Filter documents by a given geo radius using the built-in filter `_geoRadius({lat}, {lng}, {distance_in_meters})`. It is possible to cumulate several geo-search filters within the `filter` field.
```

Comment:
"Is there precedence for this in other parts of the codebase?" 

We have it for the virtual fields generated in search response like `_formatted`.

However, we have added a reserved field `_geo` within the documents to indicate the geographical coordinates of these. I naturally want to keep `_` as a prefix to the names of these new rules `_geoRadius` and `_geoPoint` to mark the fact that they are special expression elements related.

If this first iteration hits the mark. We should be able to add built-in filters rules for polygons and bounding box, it will be the moment to review the naming if we don't do it during our testing phases of this first iteration. I want to test it like that at first and return to it if necessary.

Reply:
ok, yeah, `_` having precedent makes it more solid.

Also, to add to my previous comment: python also follows the convention of `_` == private method/var/etc.

Not that I think unofficial JS/python conventions should inform our design decisions too much.